server/controllers/auth.controller.js

```diff
@@ -35,14 +37,13 @@ exports.register = async (req, res) => {
             lastName,
             email: email.toLowerCase(), // Sanitize: convert email to lowercase
             password: encryptedPassword,
-            type,
+            type: accountType[0]._id,
```

Comment:
Do you want to use the ID instead of the name

Reply:
resp is now:
{
    "status": "Success",
    "data": {
        "user": {
            "_id": 
            "firstName": 
            "lastName": 
            "email": "mmikkelson@mail.com",
            "role": {
                "_id": 
            }
        },
        "accessToken": 
    }
}